src/Annotations/Command/QueueWatchCommand.php

```diff
@@ -66,8 +68,14 @@ protected function process(InputInterface $input, QueueItem $item, $entity = nul
                 $sanitized_display_name = $display_name;
             }
             $user = new User($id, $email, $sanitized_display_name);
-            $upsert = $this->hypothesisSdk->users()->upsert($user)->wait();
-            $this->logger->info(sprintf('Hypothesis user "%s" successfully %s.', $upsert->getUsername(), ($upsert->isNew() ? 'created' : 'updated')));
+            try {
+                $upsert = $this->hypothesisSdk->users()->upsert($user)->wait();
+                $this->logger->info(sprintf('Hypothesis user "%s" successfully %s.', $upsert->getUsername(), ($upsert->isNew() ? 'created' : 'updated')));
+            } catch (BadResponse $e) {
+                // If upsert failures then log error but don't repeat.
+                $this->queue->commit($item);
+                $this->logger->error(sprintf('Hypothesis user "%s" upsert failure.', $user->getUsername()));
+            }
```

Comment:
just confirming that the parent class https://github.com/elifesciences/bus-sdk-php/blob/8e8a21ba42b8ad367e84877236109c440b6c9615/src/Command/QueueCommand.php#L97-L120 doesn't have support for committing `$item` in case of an exception, it always releases it. So this is the right place to do it, then we can pull up the logic in the library later to make it reusable.